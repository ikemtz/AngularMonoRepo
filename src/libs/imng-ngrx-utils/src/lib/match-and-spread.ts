import { findAndMerge } from "./find-and-merge";
/**
 * Spreads the objects in the dataset1, with a matching id object from the dataset2.
 * @param dataset1 The original dataset, these are the records you can expect to get back.
 * @param dataset2 This dataset will be used to augment the individuals records in dataSet1, records will be matched based on 'id'.
 * @returns dataset1
 */
export function matchAndSpread <ENTITY1 extends { id?: number | string | Date; }, ENTITY2 extends { id?: number | string | Date; }>(
  dataset1: ENTITY1[],
  dataset2: ENTITY2[]): ENTITY1[] {
    return dataset1.map(d1 => findAndMerge(d1, dataset2));
}