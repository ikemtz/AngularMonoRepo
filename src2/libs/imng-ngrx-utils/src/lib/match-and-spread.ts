import { findAndMerge } from './find-and-merge';
import { IdType } from 'imng-nrsrx-client-utils';
/**
 * Spreads the objects in the dataset1, with a matching id object from the dataset2.
 * @param dataset1 The original dataset, these are the records you can expect to get back.
 * @param dataset2 This dataset will be used to augment the individuals records in dataSet1, records will be matched based on 'id'.
 * @returns dataset1
 */
export function matchAndSpread<ENTITY1 extends { id?: IdType }, ENTITY2 extends { id?: IdType }>(
  dataset1: ENTITY1[],
  dataset2: ENTITY2[],
): ENTITY1[] {
  return dataset1.map((d1) => findAndMerge(d1, dataset2));
}
