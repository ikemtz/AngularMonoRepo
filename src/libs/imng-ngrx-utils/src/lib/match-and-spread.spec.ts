import { matchAndSpread } from './match-and-spread';

describe('matchAndSpread', () => {
  it('should work', () => {

    const dataset1 = [
      { id: 123, collection: ['😎', '🐱‍👤'] },
      { id: 456, collection: ['a', 'b', 'c'] },
    ];
    const dataset2 = [
      { id: 123, singleValue: '😎😎😉' },
      { id: 456, singleValue: '🐱', collection: ['d', 'x', 'c'] },
      { id: 789, doubeValue: '🎂', collection: ['d', 'x', 'c'] }
    ];
    const result = matchAndSpread(dataset1, dataset2);
    expect(result).toMatchSnapshot();
  });
});

