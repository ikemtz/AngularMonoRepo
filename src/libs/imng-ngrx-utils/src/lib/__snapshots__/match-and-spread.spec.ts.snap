// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchAndSpread should work 1`] = `
[
  {
    "collection": [
      "😎",
      "🐱‍👤",
    ],
    "id": 123,
    "singleValue": "😎😎😉",
  },
  {
    "collection": [
      "a",
      "b",
      "c",
    ],
    "id": 456,
    "singleValue": "🐱",
  },
]
`;
